components/measurement/nettests/WebConnectivity.js

```diff
@@ -505,19 +532,23 @@ const WebConnectivityDetails = ({
               <DetailsBox
                 title={<FormattedMessage id='Measurement.Details.Websites.DNSQueries.Heading' />}
                 content={
-                  <React.Fragment>
-                    <Flex flexWrap='wrap' mb={2}>
-                      <Box mr={1}>
-                        <strong><FormattedMessage id='Measurement.Details.Websites.DNSQueries.Label.Resolver' />:</strong>
-                      </Box>
-                      <Box>
-                        {client_resolver || '(unknown)'}
+                  Array.isArray(queries) ? (
```

Comment:
A more uniform pattern to show the `No Data` label in the boxes when keys like `queries`, `tcp_connect` are invalid or missing.